packages/sdk-core/src/ERC20Token.ts

```diff
@@ -1,22 +1,28 @@
+import {
+    ERC20WithTokenInfo,
+    ERC20WithTokenInfo__factory,
+} from "@superfluid-finance/ethereum-contracts/build/typechain";
 import { ethers } from "ethers";
 
 import Operation from "./Operation";
 import { SFError } from "./SFError";
-import ERC20WithTokenInfoABI from "./abi/ERC20WithTokenInfo.json";
 import { IBaseSuperTokenParams, ITransferFromParams } from "./interfaces";
-import { ERC20WithTokenInfo } from "./typechain/ERC20WithTokenInfo";
 import { normalizeAddress } from "./utils";
 
 export default class ERC20Token {
     readonly address: string;
     readonly contract: ERC20WithTokenInfo;
 
-    constructor(address: string) {
+    constructor(
+        address: string,
+        providerOrSigner: ethers.providers.Provider | ethers.Signer
+    ) {
         this.address = address;
-        this.contract = new ethers.Contract(
+
+        this.contract = ERC20WithTokenInfo__factory.connect(
```

Comment:
Why is it good to connect the contract here? Aren't now all the contracts possibly connected to a signer? And when write operations are invoked then that signer is not used... unless there's an error in our code and we miss passing in the new signer in which case this signer is used.

Reply:
> Why is it good to connect the contract here? Aren't now all the contracts possibly connected to a signer? 

yes, the contracts may be connected to a signer, but the proper use here is to connect a provider, so we can simply change the allowed type here if you think this is an issue.

> And when write operations are invoked then that signer is not used...

However, let's say we don't do that and we allow the signer and we throw away the signer later, there is no issue with this as you have to do this anyways if you are switching between signers in the same session-one signer at start vs. different signer later at execution is pretty similar to no signer at start then signer on first write operation and different signer on subsequent or each write operation.

> unless there's an error in our code and we miss passing in the new signer in which case this signer is used.

If the user of the product decides to execute transactions via the contract directly, this may be an issue and may warrant only allowing provider type to be passed on initialization. 
However, if they are using our class and creating an `Operation` and doing `.exec()`, this will absolutely fail even if you cast it as any because you would be trying to execute `.sendTransaction` on an undefined object.

Reply:
What's the benefit of connecting here?

Reply:
Sure that's a fair point.
Two main benefits of using this `.connect`:
Code with .connect:
```
this.contract = ERC20WithTokenInfo__factory.connect(
    this.address,
    providerOrSigner
);
```

Code without .connect:
```
this.contract = new ethers.Contract(
    this.address,
    ERC20WithTokenInfo__factory.abi
) as ERC20WithTokenInfo;
```
Some extra typecasting is avoided with `.connect`.

Secondly, for write operations, I think this will always be overridden and it is probably safe to do this. However, by providing a provider/signer in the initialization of the contract, it opens up the possibility of not requiring the end user to pass a provider on reads. This does become more problematic if the consumer is on multiple chains but this can be pretty easily solved with just passing in a new provider.

Lastly, why not, this operation is not blocking and will not slow anything down - I see no good reason to not use this (imo) cleaner way of initializing a contract.